components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/js/event-simulator/simulator.js

```diff
@@ -0,0 +1,739 @@
+ ~   You may obtain a copy of the License at
+ ~
+ ~        http://www.apache.org/licenses/LICENSE-2.0
+ ~
+ ~   Unless required by applicable law or agreed to in writing, software
+ ~   distributed under the License is distributed on an "AS IS" BASIS,
+ ~   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ ~   See the License for the specific language governing permissions and
+ ~   limitations under the License.
+ */
+
+define(['jquery', 'log', './simulator-rest-client', 'lodash', /* void libs */'bootstrap', 'theme_wso2', 'jquery_ui',
+    'jquery_validate', 'jquery_timepicker', './templates'], function ($, log, Simulator, _) {
+
+    "use strict";   // JS strict mode
+
+    var self = {};
+
+    self.init = function () {
+
+        self.singleEventConfigCount = 1;
+        self.executionPlanDetailsMap = {};
+        self.singleEventForm = $('#single-event-form').find('form').clone();
+        self.singleEventConfigTabContent = $("#single-event-config-tab-content");
+        self.singleEventConfigs = $("#single-event-configs");
+        self.addSingleEventForm = $('#add-single-event-form');
+        self.FAULTY = 'FAULTY';
+        self.STOP = 'STOP';
+        self.RUN = 'RUN';
+        self.DEBUG = 'DEBUG';
+
+        // add methods to validate int/long and double/float
+        $.validator.addMethod("validateIntOrLong", function (value, element) {
+            return this
+                    .optional(element) || /^[-+]?[0-9]+$/.test(value);
+        }, "Please provide a valid numerical value.");
+
+        $.validator.addMethod("validateFloatOrDouble", function (value, element) {
+            return this
+                    .optional(element) || /^[-+]?[0-9]*\.?[0-9]+([eE][-+]?[0-9]+)?$/.test(value);
+        }, "Please provide a valid numerical value.");
+
+        // add the first single event form and disable the delete button
+        self.addSingleEventConfigForm(null, self.addSingleEventForm);
+        $('li[data-uuid="1"] button[name="delete"]')
+            .prop('disabled', true)
+            .hide();
+
+        // add a single event form
+        self.addSingleEventForm.on('click', function (e) {
+            self.addSingleEventConfigForm(e, this);
+        });
+
+        // refresh the execution plan name drop down when execution plans are added to or removed from the workspace
+        self.singleEventConfigTabContent.on('focusin', 'select[name="execution-plan-name"]', function () {
+            var element = $(this);
+            var form = element.closest('form[data-form-type="single"]');
+            var uuid = form.data('uuid');
+            self.loadExecutionPlanNames(uuid);
+            console.log('load exc');
+        });
+
+        // When execution plan name selected changes refresh the form
+        self.singleEventConfigTabContent.on('change', 'select[name="execution-plan-name"]', function () {
+            var element = $(this);
```

Comment:
As a standard you can prefix jquery variables with "$".   

e.g. `var element = $(this);` =  `var $element = $(this);`